SERIALIZATION.md

```diff
@@ -0,0 +1,150 @@
+                .WithSerializer(serializer)
+                .Build();
+
+            Person myPerson = driver.Execute(txn =>
+            {
+                // Creating a query with native C# type parameters specifying the expected output type Person
+                IQuery<Person> myQuery = txn.Query<Person>("SELECT * FROM People WHERE Age = ?", 21);
+
+                // Execute the created query
+                IResult<Person> myResult = txn.Execute(myQuery);
+
+                foreach (var Person in myResult)
+                {
+                    return Person;
+                }
+                return null;
+            });
+
+            // myPerson and its properties are directly returned from QLDB now as C# types instead of IIonType.
+            Console.WriteLine(myPerson.Name);
+        }
+    }
+}
+```
+
+### ISerializer
+
+In the above example, this library provided an `ISerializer` to the driver, which `txn.Query` uses to create an `IQuery`. The `ObjectSerializer` is a wrapper around the [Amazon Ion Object Mapper](https://github.com/amzn/ion-object-mapper-dotnet) with default settings which supports the following [primitive type conversions](https://github.com/amzn/ion-object-mapper-dotnet/blob/main/SPEC.md#primitive-type-conversion). If desired, a custom implementation can be used as long as it implements the `ISerializer` interface.
+
+```c#
+public interface ISerializer
+{
+    /// <summary>
+    /// Serialize a C# object into a ValueHolder object containing the Ion binary or text value.
+    /// </summary>
+    ///
+    /// <param name="o">The C# object to be serialized into ValueHolder.</param>
+    ///
+    /// <returns>The ValueHolder object containing the Ion binary or text value.</returns>
+    ValueHolder Serialize(object o);
+
+    /// <summary>
+    /// Deserialize a ValueHolder object containing the Ion binary or text value into an object of type T.
+    /// </summary>
+    ///
+    /// <param name="s">The ValueHolder object to be deserialized into an object of type T.</param>
+    /// <typeparam name="T">The return type.</typeparam>
+    ///
+    /// <returns>The object of type T.</returns>
+    T Deserialize<T>(ValueHolder s);
+}
+```
+
+However, as implementing a fully-functioning `ISerializer` is challenging, it is *highly recommended* to instead utilize the `IonSerializer` in the Amazon Ion Object Mapper mentioned above. It is highly configurable and can be wrapped to call it's own `Serialize` and `Deserialize<T>` methods to satisfy the interface of a custom `Iserializer`.
+
+### IQuery
+
+In the Basic Usage Example, it is shown that `txn.Execute<T>(IQuery<T> query)` is used to return an `IResult<T>` where `T` is the type determined by the `IQuery` parameter. `txn.Query` was used to generate this, using the `ISerializer` that was passed into the builder for the driver. If a different `ISerializer` is desired for a specific `IQuery`, it is possible to do so.
+
+```c#
```

Comment:
Probably worth calling out in a comment in the code what's different between the prior example and this one, I had to scroll up and down to see it.